op-node/rollup/derive/engine_queue.go

```diff
@@ -705,6 +705,21 @@ func (eq *EngineQueue) Reset(ctx context.Context, _ eth.L1BlockRef, _ eth.System
 	if err := eq.safeHeadNotifs.SafeHeadReset(safe); err != nil {
 		return err
 	}
+	if eq.cfg.Genesis.L2.Number == 0 && safe.Hash == eq.cfg.Genesis.L2.Hash {
+		// The L2 genesis/bedrock activation block is always safe by definition
```

Comment:
I think we've very badly abused the term genesis which is a bit unfortunate.  The bedrock activation block is very definitely not genesis (it has a parent block and isn't the start of the chain history) but in a lot of places we do call it the genesis block. That's caused all kinds of confusion and bugs with fault proofs at different times.

That said, the ship seems to have well and truly sailed since the rollup config refers to the bedrock activation block as genesis so have switched this to say `rollup genesis` as suggested.  We just have to always remember that genesis is not always genesis.